lib/topological_inventory/ansible_tower/operations/core/ansible_tower_client.rb

```diff
@@ -0,0 +1,129 @@
+require "topological_inventory/ansible_tower/logging"
+require "topological_inventory/ansible_tower/connection"
+require "topological_inventory/ansible_tower/operations/core/sources_api_client"
+require "topological_inventory/ansible_tower/operations/core/topology_api_client"
+
+module TopologicalInventory
+  module AnsibleTower
+    module Operations
+      module Core
+        class AnsibleTowerClient
+          include Logging
+          include Core::TopologyApiClient
+
+          attr_accessor :connection_manager
+
+          SLEEP_POLL = 10
+          POLL_TIMEOUT = 1800
+
+          def initialize(source_id, task_id, identity = nil)
+            self.identity   = identity
+            self.source_id  = source_id
+            self.task_id    = task_id
+
+            self.connection_manager = TopologicalInventory::AnsibleTower::Connection.new
+          end
+
+          # Format of order params (Input for Catalog - created by Collector, Output is produced by catalog - input of this worker)
+          #
+          # @example:
+          #
+          # * Input (ServicePlan.create_json_schema field)(created by lib/topological_inventory/ansible_tower/parser/service_plan.rb)
+          #     {"schema":
+          #       {"fields":[
+          #         {"name":"providerControlParameters", ... },
+          #         {"name":"NAMESPACE", "type":"text", ...},
+          #         {"name":"MEMORY_LIMIT","type":"text","default":"512Mi","isRequired":true,...},
+          #         {"name":"POSTGRESQL_USER",type":"text",...},
+          #         ...
+          #        ]
+          #       },
+          #      "defaultValues":{"NAMESPACE":"openshift","MEMORY_LIMIT":"512Mi","POSTGRESQL_USER":"","VOLUME_CAPACITY":"...}
+          #     }
+          #
+          # * Output (== @param **order_params**):
+          #
+          #     { "NAMESPACE":"openshift",
+          #       "MEMORY_LIMIT":"512Mi",
+          #       "POSTGRESQL_USER":"",
+          #       "providerControlParameters":{"namespace":"default"},
+          #       ...
+          #     }"
+          def order_service_plan(job_template_id, order_params)
+            job_template = ansible_tower.api.job_templates.find(job_template_id)
+
+            job = job_template.launch(job_values(order_params))
+
+            # This means that api_client:job_template.launch() called job.find(nil), which returns list of jobs
+            # => status error was returned, but api_client doesn't return errors
+            raise ::AnsibleTowerClient::ResourceNotFoundError, "Job not found" if job.respond_to?(:count)
```

Comment:
Wow that's nasty, something we can fix in the client gem (not holding this up, just for the future)?

Reply:
yes, I suppose there should be more work on client, because its support for api v2 is limited

Reply:
it works like `find(nil)` launches the same code like `all()`